tutorial5.md

```diff
@@ -0,0 +1,337 @@
+# Tutorial 5 (controller-manager)
+
+[`kube-controller-manager`][cm] is responsible for synchronizing state between api server and the real world. 
+The API server is responsible for recording the desired state of the cluster, then the controller manager is responsible for actualizing it.
+
+As the name suggests, the controller manager is not just one application. 
+Rather, it's a single binary with a number of different components built into it.
+
+The [`kube-controller-manager` documentation][cmcli] lists the following controller managers:
+
+* `attachdetach`
+* `bootstrapsigner`
+* `clusterrole-aggregation`
+* `cronjob`
+* `csrapproving`
+* `csrcleaner`
+* `csrsigning`
+* `daemonset`
+* `deployment`
+* `disruption`
+* `endpoint`
+* `garbagecollector`
+* `horizontalpodautoscaling`
+* `job`
+* `namespace`
+* `nodeipam`
+* `nodelifecycle`
+* `persistentvolume-binder`
+* `persistentvolume-expander`
+* `podgc`
+* `pv-protection`
+* `pvc-protection`
+* `replicaset`
+* `replicationcontroller`
+* `resourcequota`
+* `root-ca-cert-publisher`
+* `route`
+* `service`
+* `serviceaccount`
+* `serviceaccount-token`
+* `statefulset`
+* `tokencleaner`
+* `ttl`
+* `ttl-after-finished`
+
+We can't go through all of these, so we'll just concentrate on a few.
+
+[cm]: https://kubernetes.io/docs/concepts/overview/components/#kube-controller-manager
+[cmcli]: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/
+
+## Credentials
+
+`kube-controller-manager` accesses the API server. 
+Like all other clients, we'll generate a `kubeconfig` so the controller-manager can access the API server.
+
+```yaml
+$ sudo kubeadm init phase kubeconfig controller-manager
+[kubeconfig] Writing "controller-manager.conf" kubeconfig file
+```
+
+This will be written to `/etc/kubernetes/controller-manager.conf`
+
+## The `kube-controller-manager` pod
+
+Just like before, we'll start out with a pod skeleton. 
+The kubeadm command to generate a fully-generated manifest is `sudo kubeadm init phase control-plane controller-manager`.
```

Comment:
the break here isn't clear, I wound up accidentally running this command rather than following the instructions. -- may want to say "but we'll be walking you through constructing the file by hand as usual."